### ISSUE TITLE
Firebase cloud messaging recipient local time zone on ios does not work

### ISSUE THREAD

**Issue by gilgil28** [firebase/quickstart-unity#340]
## Please fill in the following fields:
Unity editor version: 2018.3.8f1
Firebase Unity SDK version: 5.6.0
Platform you are targeting (iOS, Android, and/or desktop):  ios/android

## Please describe the issue here:
(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)
It seems like you had a bug on ios that prevented from sending push notifications on local time zone to work. According to the ios sdk release note of version 5.11.0, that was fixed:
**Fixed an issue where messages failed to be delivered to the recipient's time zone. (#1946)**
If the versioning is consistent across platforms, I guess that a version 5.11.0 fix on ios sdk is not included in version 5.6.0 on unity (which contains version 5.6.0 of ios and android?).
How can I fix that? It is a very important feature for us.

**Comment by chkuang-g** [comment]
Hi @gilgil28 

Thank you for reporting this issue. 
iOS sdk was released last October so it does not make sense that Unity SDK 5.6.0 did not include this fix.

Does this start to happen after you upgrade to Unity SDK 5.6.0?  Or does it happen to the previous version as well?  Which Unity SDK version did you upgrade from?

Also, it would be very helpful if you can provide reproduction steps using our Unity quickstart.
https://github.com/firebase/quickstart-unity/tree/master/messaging/testapp
This would help us to narrow down the potential issue.

Thank you

**Comment by gilgil28** [comment]
@chkuang-g thank you for your reply.
So I understand that there is no correlation between Unity SDK versioning and iOS SDK versioning?
We were using Unity SDK  5.4.1 on which our problem happened. We were wondering whether upgrading to 5.6.0 would make any sense, since the release notes do not refer to the standalone iOS\android SDKs changes that are supposedly an integrated part within.

So my question is - Does Unity SDK 5.6.0 released on March 18th contain all iOS SDK fixes until and including iOS SDK 5.18.2 released on March 7th?

**Comment by patm1987** [comment]
If you look at `Assets/Firebase/Editor/MessagingDependencies`, the Unity 5.6.0 SDK should give you
```
  <iosPods>
    <iosPod name="Firebase/Messaging" version="5.18.0" minTargetSdk="8.0">
    </iosPod>
  </iosPods>
```

You can try tweaking that and see if it fixes your issue (if you look under the generated Xcode workspace, that should correspond to the generated Podfile).

**Comment by chkuang-g** [comment]
@gilgil28 

Firebase FCM Unity SDK wraps iOS SDK.  Firebase SDK 5.6.0 is using iOS SDK 5.18.0, as @patm1987 pointed out, which should include the fix in iOS SDK 5.11.0.

When you said "prevented from sending push notifications on local time zone to work", do you mean the client never received any message or they received it immediately, but not at the specified time in their time zone?

**Comment by gilgil28** [comment]
@chkuang-g They never received the message. This is relevant for Unity SDK 5.4.1.
As I can see in MessagingDependencies.xml, Unity SDK 5.4.1 is using version 5.9.0 which is before the fix. That should answer my question. 
Nevertheless, I think this should be more clear when reading the Unity SDK release notes.

**Comment by chkuang-g** [comment]
That is good suggestion. I will forward to my team about the release note.

I think you should upgrade to Unity SDK 5.6.0, which includes iOS SDK 5.18.0.  Please let us know if this resolve your issue.

Thank you.

**Comment by ryanmeier** [comment]
@gilgil28 
I'm going to close this issue since it seems like upgrading the SDK will resolve the problem you're seeing.  If you run into any more issues, feel free to let us know!

**Comment by gilgil28** [comment]
@ryanmeier @chkuang-g 
OK, I upgraded the unity SDK to version 5.6.0 and the problem is not resolved. 
Using scheduled push notification with local time zone is not working as the users are not getting any push notification at all (iOS only!)

**Comment by ryanmeier** [comment]
@gilgil28 
Sorry to hear you're still running into this problem!  

To clarify - are you able to reproduce this behavior in our messaging testapp? (https://github.com/firebase/quickstart-unity/tree/master/messaging/testapp)

If so, can you share the steps you're taking which allow you to reproduce the issue?  As @chkuang-g mentioned, that would be very helpful for us to be able to more easily narrow down where the problem lies.

**Comment by chliangGoogle** [comment]
@gilgil28 The (#1946) only expands for certain platform timezone return nil issue, should not be a generic cause/fix for yours. 

How do you tell user is not able to receive it at all?  Are you seeing from analytics dashboard? Are you able to reproduce the issue?

**Comment by chkuang-g** [comment]
@gilgil28 
Any update to this issue?

**Comment by chliangGoogle** [comment]
@gilgil28 another question, do you use scheduled message or just "send now" message?

**Comment by chliangGoogle** [comment]
Close this for now but please do reopen if you can reproduce with the latest SDK. Thank you!

**Comment by ed-yaku** [comment]
Hi @chliangGoogle I would like to contribute to this issue as I have also been experiencing it. I'll add all required details below along with reproduction steps.

**Unity editor version**: 2018.1.0f2
**Firebase Unity SDK version**: 6.1.0
Platform you are targeting (iOS, Android, and/or desktop): **ios/android**

We have found that if we send notifications from the Firebase Cloud Messaging console using scheduled **Recipient time zone** then we receive the notifications on Android devices but not on iOS.

We recently upgraded from Firebase Unity SDK 4.4.2 to 6.1.0. I also noticed the already mentioned bug fix (#1946) so was hoping our issue would be fixed but that hasn't been the case.

<h2>Reproduction Steps:</h2>

1. Start with a clean Unity project and import the Messaging unity package (.NET 3).
2. Add this simple initialisation script (
[TestCloudMessaging.cs.zip](https://github.com/firebase/quickstart-unity/files/3345130/TestCloudMessaging.cs.zip)
) component to a GameObject in the SampleScene.
3. Build the SampleScene for iOS.
4. Start the app to accept the notification permissions.
5. Now set **Settings > General > Date & Time**: to **"Kiritimati, Kiribate (UTC+14)"**. This is for ease of testing as scheduled local time notifications need to be in the future everywhere.
6. In Firebase Cloud Messaging composer create a new notification. In **Scheduling** make sure you use **Recipient time zone** and set the time slightly ahead of your device time that is now on UTC+14
7. Publish message.
8. Message does not appear on device at the scheduled time.

This functionality was tested on Android (NB: used Tonga Standard Time UTC+13) and the notification came through to device.

If you would like any other information please shout.

Thanks
